Dockerfile.alpine

```diff
@@ -0,0 +1,46 @@
+FROM alpine:3.14
+
+ARG openssl_arch=linux-x86_64
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH \
+    DEBIAN_FRONTEND=noninteractive
+
+RUN apk add --no-cache curl wget linux-headers make perl makedepend \
+    grep coreutils binutils linux-lts-dev linux-lts libxml2-dev gcc libc-dev \
+    clang-libs llvm11 llvm11-libs llvm11-dev llvm11-static g++
+
+RUN curl https://sh.rustup.rs -sSf > rustup.sh \
+    && sh rustup.sh -y \
+          --default-toolchain stable \
+          --profile minimal \
+          --no-modify-path \
+    && rustup component add rustfmt \
+    && rustup target add x86_64-unknown-linux-musl \
+    && rustup toolchain add nightly --profile minimal \
+    && rustup --version \
+    && cargo --version \
+    && rustc --version \
+    && rustup install 1.51 \
+    && cargo install bindgen
+
+RUN echo "Building OpenSSL" \
+    && VERS=1.0.2p \
+    && CHECKSUM=50a98e07b1a89eb8f6a99477f262df71c6fa7bef77df4dc83025a2845c827d00 \
+    && curl -sqO https://www.openssl.org/source/openssl-$VERS.tar.gz \
+    && echo "$CHECKSUM  openssl-$VERS.tar.gz" > checksums.txt \
+    && sha256sum -c checksums.txt \
+    && tar xzf openssl-$VERS.tar.gz && cd openssl-$VERS \
+    && ./Configure ${openssl_arch} -fPIC --prefix=/usr/local/musl/$TARGET \
+    && make depend \
+    && make && make install \
+    && cd .. && rm -rf openssl-$VERS.tar.gz openssl-$VERS checksums.txt
```

Comment:
Since Alpine is musl anyway i think it's better to use the openssl version they ship in APK.

Reply:
By the way, why is musl installed in ubuntu or fedora?

Reply:
Since we generally want to be able to test building a static binary on those platforms, too